enterprise/server/backends/authdb/authdb.go

```diff
@@ -537,11 +539,12 @@ func (d *AuthDB) createAPIKey(db *db.DB, userID, groupID, label string, caps []a
 		return nil, err
 	}
 	keyPerms := int32(0)
-	if userID == "" {
+	if ak.UserID == "" {
 		keyPerms = perms.GROUP_READ | perms.GROUP_WRITE
 	} else {
 		keyPerms = perms.OWNER_READ | perms.OWNER_WRITE
 	}
+	ak.Perms = keyPerms
```

Comment:
nit: consider cloning the API key or passing it by value, since it's easy to think that this returns the newly created key without modifying the original

```suggestion
	{
		clone := *ak
		ak = *clone
	}
	ak.Perms = keyPerms
```

Reply:
Updated to pass by value.